packages/lesswrong/lib/editor/make_editable.ts

```diff
@@ -94,16 +97,115 @@ const defaultOptions: MakeEditableOptions = {
 
 export const editableCollections = new Set<CollectionNameString>()
 export const editableCollectionsFields: Record<CollectionNameString,Array<string>> = {} as any;
-export const editableCollectionsFieldOptions: Record<CollectionNameString,Record<string,MakeEditableOptions>> = {} as any;
+export const editableCollectionsFieldOptions: Record<CollectionNameString, Record<string, MakeEditableOptions<CollectionNameString>>> = {} as any;
 let editableFieldsSealed = false;
 export function sealEditableFields() { editableFieldsSealed=true }
 
+const buildEditableResolver = <N extends CollectionNameString>(
+  collection: CollectionBase<N>,
+  fieldName: EditableFieldName<N>,
+  normalized: boolean,
+): CollectionFieldResolveAs<N> => {
+  if (normalized) {
+    // TODO: Handle explicit versions
+    return {
+      type: "Revision",
+      arguments: "version: String",
+      resolver: async (
+        doc: ObjectsByCollectionName[N],
+        {version}: {version?: string},
+        context: ResolverContext,
+      ): Promise<DbRevision|null> => {
+        const {currentUser, Revisions} = context;
+        const {checkAccess} = Revisions;
+        const revision = await Revisions.findOne({
+          _id: doc[`${fieldName}_latest` as keyof ObjectsByCollectionName[N]],
+        });
+        return revision && await checkAccess(currentUser, revision, context)
+          ? revision
+          : null;
+      },
+      sqlResolver: ({field, resolverArg, join}) => join({
+        table: "Revisions",
+        type: "left",
+        on: {
+          _id: field(`${fieldName}_latest` as FieldName<N>),
+        },
+        resolver: (revisionField) => revisionField("*"),
+      }),
+    };
+  }
+
+  return {
```

Comment:
This denormalized resolver is copied from below - we should be able to remove it if you end up normalizing all editable fields.